### DISCUSSION TITLE
Fastapi Deployment

### DISCUSSION THREAD

**Discussion by joachimhodana** [vercel/community#3721] [Help] [answered]
### Summary

Hey so I'm trying to deploy my API to vercel, the issue is that it's not working. What i think is the issue is that Vercel doesn't use my running command or something. My API code: https://github.com/Style77/Parkr-api/tree/develop, as you can see everything is in `parkr` directory, and to start my api i just use `python -m parkr` (like module). On local everything is working, this is my vercel config

```json
{
  "devCommand": "python3 -m parkr",
  "builds": [
    {
      "src": "parkr",
      "use": "@vercel/python"
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "parkr"
    }
  ]
}
```
I also tried using `@now/python`, src set to `parkr/__main__.py` (with and without that `devCommand`, but nothing works.

To try my api I'm going to https://parkr-api-style77.vercel.app/docs, but as you can see it does return 404 error.

What's wrong?

### Example

https://parkr-api-style77.vercel.app/docs

### Steps to Reproduce

Clone my repo, create file `vercel.json` paste my config and run `npx vercel .`

**Comment by amyegan** [comment, marked as answer]
Hey @Style77. Serverless environments require slightly different setup from traditional servers. The following resources should help you get your site configured and deployed:

- [Using the Python Runtime with Vercel Serverless Functions](https://vercel.com/docs/functions/serverless-functions/runtimes/python#)
- [Python starter templates and themes](https://vercel.com/templates/python)
  - [Python Hello World](https://github.com/vercel/examples/tree/main/python/hello-world)
  - [Next.js FastAPI Starter](https://github.com/digitros/nextjs-fastapi)
  - [Flask Alt Text Generator](https://github.com/Nutlope/alt-text-generator)

**Reply by joachimhodana** [reply]
I did read that articles, but they didn't really help me, as in my first comment the issue is that Im using a lot of files for declaring routes, and my api is considered as python module - thats why i run it with python -m parkr, however i got no idea how to do run it with serverless functions.

**Reply by amyegan** [reply]
Please refer to the the example repos to get a better understanding of how to set up your project.
